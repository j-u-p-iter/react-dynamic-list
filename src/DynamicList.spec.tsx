import * as React from "react";
import { fireEvent, render } from "react-testing-library";

import { DynamicList as DynamicListController } from ".";

describe("DynamicList", () => {
  let getByText: any;
  let container: any;

  beforeAll(() => {
    const dataItems = [
      {
        title: "Some Title"
      },
      {
        title: "One More Title"
      }
    ];

    const DynamicList = () => (
      <DynamicListController data={dataItems}>
        {({ api: { addItem, removeItem, getItems } }) => {
          const items = getItems();

          return items ? (
            <div>
              <ul>
                {items.map(({ title }, index) => (
                  <li key={index}>
                    {title}
                    <button
                      onClick={() => removeItem(index)}
                    >{`Remove Item ${title}`}</button>
                  </li>
                ))}
              </ul>
              <button onClick={() => addItem({ title: "And One More Item" })}>
                Add Item
              </button>
            </div>
          ) : null;
        }}
      </DynamicListController>
    );

    ({ container, getByText } = render(<DynamicList />));
  });

  it("exposes correct API", () => {
    expect(container.firstChild).toMatchSnapshot();

    fireEvent.click(getByText("Add Item"));

    expect(container.firstChild).toMatchSnapshot();

    fireEvent.click(getByText("Remove Item Some Title"));

    expect(container.firstChild).toMatchSnapshot();
  });
});
