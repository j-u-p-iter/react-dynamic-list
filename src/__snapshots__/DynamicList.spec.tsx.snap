// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicList exposes correct API 1`] = `
<div>
  <ul>
    <li>
      Some Title
      <button>
        Remove Item Some Title
      </button>
    </li>
    <li>
      One More Title
      <button>
        Remove Item One More Title
      </button>
    </li>
  </ul>
  <button>
    Add Item
  </button>
  <button>
    Add Item Without Callback
  </button>
  <button>
    Set new items
  </button>
</div>
`;

exports[`DynamicList exposes correct API 2`] = `
<div>
  <ul>
    <li>
      Some Title
      <button>
        Remove Item Some Title
      </button>
    </li>
    <li>
      One More Title
      <button>
        Remove Item One More Title
      </button>
    </li>
    <li>
      And One More Item
      <button>
        Remove Item And One More Item
      </button>
    </li>
  </ul>
  <button>
    Add Item
  </button>
  <button>
    Add Item Without Callback
  </button>
  <button>
    Set new items
  </button>
</div>
`;

exports[`DynamicList exposes correct API 3`] = `
<div>
  <ul>
    <li>
      One More Title
      <button>
        Remove Item One More Title
      </button>
    </li>
    <li>
      And One More Item
      <button>
        Remove Item And One More Item
      </button>
    </li>
  </ul>
  <button>
    Add Item
  </button>
  <button>
    Add Item Without Callback
  </button>
  <button>
    Set new items
  </button>
</div>
`;

exports[`DynamicList exposes correct API 4`] = `
<div>
  <ul>
    <li>
      New Some Title
      <button>
        Remove Item New Some Title
      </button>
    </li>
    <li>
      New One More Title
      <button>
        Remove Item New One More Title
      </button>
    </li>
  </ul>
  <button>
    Add Item
  </button>
  <button>
    Add Item Without Callback
  </button>
  <button>
    Set new items
  </button>
</div>
`;
